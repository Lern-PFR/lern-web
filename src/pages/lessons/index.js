import LessonCreationPage from './LessonCreationPage';
import LessonEditionPage from './LessonEditionPage';

export {
	LessonCreationPage,
	LessonEditionPage,
};
