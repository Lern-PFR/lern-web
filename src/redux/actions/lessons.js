import * as LessonsApi from 'api/lessonsApi';
import i18next from 'i18next';
import { toast } from 'react-toastify';
import { fetchConcept } from './concepts';

/**
 * @constant
 * @name ActionTypes
 * @description The various action types used to interact with the lessons redux state.
 * @type {object}
 *
 * @author Timothée Simon-Franza
 */
export const ActionTypes = {
	FETCH_LESSON_REQUEST: '@LESSONS/FETCH_REQUEST',
	FETCH_LESSON_SUCCESS: '@LESSONS/FETCH_SUCCESS',
	FETCH_LESSON_FAILURE: '@LESSONS/FETCH_FAILURE',

	CREATE_LESSON_REQUEST: '@LESSONS/CREATE_REQUEST',
	CREATE_LESSON_SUCCESS: '@LESSONS/CREATE_SUCCESS',
	CREATE_LESSON_FAILURE: '@LESSONS/CREATE_FAILURE',

	UPDATE_LESSON_REQUEST: '@LESSONS/UPDATE_REQUEST',
	UPDATE_LESSON_SUCCESS: '@LESSONS/UPDATE_SUCCESS',
	UPDATE_LESSON_FAILURE: '@LESSONS/UPDATE_FAILURE',

	DELETE_LESSON_REQUEST: '@LESSONS/DELETE_REQUEST',
	DELETE_LESSON_SUCCESS: '@LESSONS/DELETE_SUCCESS',
	DELETE_LESSON_FAILURE: '@LESSONS/DELETE_FAILURE',

	CLEAR_LESSON_LIST: '@LESSONS/CLEAR_LIST',
};

// //////////////////////////////////////////////////////// //
// //////////// Specific lesson fetching actions ////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name fetchLessonRequest
 * @description Action triggered anytime a lesson fetching call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const fetchLessonRequest = () => ({ type: ActionTypes.FETCH_LESSON_REQUEST });

/**
  * @function
  * @name fetchLessonSuccess
  * @description Action triggered as a result to a successful lesson fetching API call.
  *
  * @author Timothée Simon-Franza
  *
  * @param {object} lesson : The lesson retrieved from the API.
  *
  * @returns {object}
  */
const fetchLessonSuccess = ({ lesson }) => ({
	type: ActionTypes.FETCH_LESSON_SUCCESS,
	payload: { lesson },
});

/**
 * @function
 * @name fetchLessonFailure
 * @description Action triggered as a result to a failed lesson fetching API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const fetchLessonFailure = (error) => ({
	type: ActionTypes.FETCH_LESSON_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// //////////////// Lesson creation actions /////////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name createLessonRequest
 * @description Action triggered anytime a lesson creation call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const createLessonRequest = () => ({ type: ActionTypes.CREATE_LESSON_REQUEST });

/**
 * @function
 * @name createLessonSuccess
 * @description Action triggered as a result to a successfult lesson creation API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} lesson : The newly created lesson.
 *
 * @returns {object}
 */
const createLessonSuccess = ({ lesson }) => ({
	type: ActionTypes.CREATE_LESSON_SUCCESS,
	payload: { lesson },
});

/**
 * @function
 * @name createLessonFailure
 * @description Action triggered as a result to a failed lesson creation API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const createLessonFailure = (error) => ({
	type: ActionTypes.CREATE_LESSON_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// //////////////// Lesson edition actions //////////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name updateLessonRequest
 * @description Action triggered anytime a lesson edition call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const updateLessonRequest = () => ({ type: ActionTypes.UPDATE_LESSON_REQUEST });

/**
 * @function
 * @name updateLessonSuccess
 * @description Action triggered as a result to a successful lesson edition API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} lesson : The updated lesson object.
 *
 * @returns {object}
 */
const updateLessonSuccess = ({ lesson }) => ({
	type: ActionTypes.UPDATE_LESSON_SUCCESS,
	payload: { lesson },
});

/**
 * @function
 * @name updateLessonFailure
 * @description Action triggered as a result to a failed lesson edition API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const updateLessonFailure = (error) => ({
	type: ActionTypes.UPDATE_LESSON_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// //////////////// Lesson deletion actions //////////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name deleteLessonRequest
 * @description Action triggered anytime a lesson deletion call is made to the API.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const deleteLessonRequest = () => ({ type: ActionTypes.DELETE_LESSON_REQUEST });

/**
 * @function
 * @name deleteLessonSuccess
 * @description Action triggered as a result to a successful lesson deletion API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} lesson : The deleted lesson object.
 *
 * @returns {object}
 */
const deleteLessonSuccess = ({ lesson }) => ({
	type: ActionTypes.DELETE_LESSON_SUCCESS,
	payload: { lesson },
});

/**
 * @function
 * @name deleteLessonFailure
 * @description Action triggered as a result to a failed lesson deletion API call.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} error : The exception sent back from the API.
 *
 * @returns {object}
 */
const deleteLessonFailure = (error) => ({
	type: ActionTypes.DELETE_LESSON_FAILURE,
	payload: { error },
});

// //////////////////////////////////////////////////////// //
// ///////////// Lesson list clearing actions ///////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name clearLessonListAction
 * @description Action triggered anytime the lesson list is not needed anymore.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {object}
 */
const clearLessonListAction = () => ({ type: ActionTypes.CLEAR_LESSON_LIST });

// //////////////////////////////////////////////////////// //
// //////////////// Exported action creators ////////////// //
// //////////////////////////////////////////////////////// //

/**
 * @function
 * @name fetchLesson
 * @description Method used to fetch the latest version of a specified lesson.
 *
 * @author Timothée Simon-Franza
 *
 * @param {string} lessonId : The id of the lesson we want to retrieve.
 */
export const fetchLesson = (lessonId) => (dispatch) => {
	dispatch(fetchLessonRequest());

	return LessonsApi.fetchLessonById(lessonId)
		.then(({ lesson }) => dispatch(fetchLessonSuccess({ lesson })))
		.catch((error) => dispatch(fetchLessonFailure(error)));
};

/**
 * @function
 * @name createLesson
 * @description Method used to create a new lesson in the database.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} lessonData : The data to create the new lesson from.
 */
export const createLesson = (lessonData) => (dispatch) => {
	dispatch(createLessonRequest());

	return LessonsApi.createLesson(lessonData)
		.then(({ lesson }) => dispatch(createLessonSuccess({ lesson })))
		.catch((error) => dispatch(createLessonFailure(error)));
};

/**
 * @function
 * @name updateLesson
 * @description Method used to update an existing lesson instance from the database.
 *
 * @author Timothée Simon-Franza
 *
 * @param {object} lessonData	: The data to update the lesson with.
 * @param {string} lessonId		: The id to identify the lesson to update.
 */
export const updateLesson = (lessonData, lessonId) => (dispatch) => {
	dispatch(updateLessonRequest());

	return LessonsApi.updateLesson(lessonData, lessonId)
		.then(({ lesson }) => dispatch(updateLessonSuccess({ lesson })))
		.catch((error) => dispatch(updateLessonFailure(error)));
};

/**
 * @function
 * @name deleteLesson
 * @description Method used to remove an existing lesson from the database.
 *
 * @author Timothée Simon-Franza
 *
 * @param {string} lessonId : The id identifying the lesson to remove.
 */
export const deleteLesson = (lessonId) => (dispatch) => {
	dispatch(deleteLessonRequest());

	return LessonsApi.deleteLesson(lessonId)
		.then((lesson) => {
			dispatch(deleteLessonSuccess({ lesson }));
			toast.success(i18next.t('lessons.deletion.toasts.success', { name: lesson.title }));
			dispatch(fetchConcept(lesson.conceptId));
		})
		.catch((error) => dispatch(deleteLessonFailure(error)));
};

/**
 * @function
 * @name clearLessonList
 * @description Method used to clear the lesson list stored in memory when the application does not need it anymore.
 *
 * @author Timothée Simon-Franza
 *
 * @returns {Promise.resolve}
 */
export const clearLessonList = () => (dispatch) => {
	dispatch(clearLessonListAction());

	return Promise.resolve();
};
