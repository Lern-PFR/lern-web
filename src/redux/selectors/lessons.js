import { sortBy } from 'lodash';
import { createSelector } from 'reselect';
import { getUserAnswer } from './userAnswers';

// const extractFirstQuestionFromLesson = (lesson) => (lesson?.exercises?.[0]?.questions?.[0]);

/**
 * @function
 * @name extractFirstQuestionFromLesson
 * @description Extracts the first question of the given lesson if it exists.
 *
 * @param {object} lesson The lesson to extract a question from.
 *
 * @return {object|undefined}
 */
const extractFirstQuestionFromLesson = createSelector(
	getUserAnswer,
	(_, lesson) => lesson ?? undefined,
	(userAnswer, lesson) => {
		const question = lesson?.exercises?.[0]?.questions?.[0];

		if (!question) {
			return undefined;
		}

		if (question && userAnswer && question.id === userAnswer.questionId && question.answers) {
			question.answers = question.answers.map((answer) => ({
				...answer,
				isUserAnswer: answer.id === userAnswer.answerId,
			}));
		}

		return {
			...question,
			answers: sortBy(question.answers ?? [], 'text'),
		};
	}
);

export {
	// eslint-disable-next-line import/prefer-default-export
	extractFirstQuestionFromLesson,
};
