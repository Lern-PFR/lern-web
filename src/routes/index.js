import keys from './keys';

export {
	keys as default,
};
