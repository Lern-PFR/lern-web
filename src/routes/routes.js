/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/display-name */
import { PostSignupPage, SignInPage, SignupPage } from 'pages/auth';
import { ConceptCreationPage, ConceptDetailsPage, ConceptEditionPage } from 'pages/concepts';
import { SubjectCreationPage, SubjectDetailsPage, SubjectEditionPage, SubjectListPage, SubjectStructurePage } from 'pages/subjects';
import { ModuleCreationPage, ModuleDetailsPage, ModuleEditionPage } from 'pages/modules';
import { Homepage } from 'pages/home';
import { LessonCreationPage } from 'pages/lessons';
import keys from './keys';

/**
 * Routes to render
 * For authentication purposes, each route can have `allowAnonymous` and `allowAuthenticated` values
 * Default values are :
 *  - `allowAnonymous`		: false
 *  - `allowAuthenticated`	: true
 */
export default [
	{
		path: keys.app.default,
		exact: true,
		allowAnonymous: true,
		component: Homepage,
	},
	{
		path: keys.about.default,
		exact: true,
		allowAnonymous: true,
		allowAuthenticated: true,
		component: () => <p>About</p>,
	},
	{
		path: keys.contactUs.default,
		exact: true,
		allowAnonymous: true,
		allowAuthenticated: true,
		component: () => <p>Contact us</p>,
	},
	// Home
	{
		path: keys.home.default,
		exact: true,
		allowAnonymous: true,
		allowAuthenticated: true,
		component: Homepage,
	},
	// Auth
	{
		path: keys.auth.default,
		routes: [
			{
				path: keys.auth.login,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: SignInPage,
			},
			{
				path: keys.auth.logout,
				exact: true,
				component: () => <p>Logout</p>,
			},
			{
				path: keys.auth.forgottenPassword,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: () => <p>Forgot password</p>,
			},
			{
				path: keys.auth.signup,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: SignupPage,
			},
			{
				path: keys.auth.postSignup,
				allowAnonymous: true,
				allowAuthenticated: false,
				exact: true,
				component: PostSignupPage,
			},
		],
	},
	// Subjects
	{
		path: keys.subjects.default,
		routes: [
			{
				path: keys.subjects.list,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectListPage,
			},
			{
				path: keys.subjects.subjectStructure,
				allowAnonymous: true,
				allowAuthenticated: true,
				exact: true,
				component: SubjectStructurePage,
			},
			{
				path: keys.subjects.subjectCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectCreationPage,
			},
			{
				path: keys.subjects.subjectEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectEditionPage,
			},
			{
				path: keys.subjects.moduleCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ModuleCreationPage,
			},
			{
				path: keys.subjects.subjectDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: SubjectDetailsPage,
			},
		],
	},
	// Modules
	{
		path: keys.modules.default,
		routes: [
			{
				path: keys.modules.moduleEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ModuleEditionPage,
			},
			{
				path: keys.modules.conceptCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptCreationPage,
			},
			{
				path: keys.modules.moduleDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ModuleDetailsPage,
			},
		],
	},
	// Concepts
	{
		path: keys.concepts.default,
		routes: [
			{
				path: keys.concepts.conceptEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptEditionPage,
			},
			{
				path: keys.concepts.conceptDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptDetailsPage,
			},
			{
				path: keys.concepts.lessonCreation,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: LessonCreationPage,
			},
			{
				path: keys.concepts.lessonDetails,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: ConceptDetailsPage,
			},
		],
	},
	// Lessons
	{
		path: keys.lessons.default,
		routes: [
			{
				path: keys.lessons.lessonEdition,
				allowAnonymous: false,
				allowAuthenticated: true,
				exact: true,
				component: () => <p>Lesson edition</p>,
			},
		],
	},
];
