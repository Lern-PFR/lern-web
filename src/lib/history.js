import { createBrowserHistory } from 'history';
import { clearAlerts } from '../redux/actions/alerts';
import store from '../redux/store/configureStore';

const { dispatch } = store;

// Init applicationPathname
const history = createBrowserHistory();

history.listen(() => {
  dispatch(clearAlerts());
});

export default history;
