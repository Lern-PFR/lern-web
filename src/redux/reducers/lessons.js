import { ActionTypes } from 'redux/actions/lessons';

const initialState = {
	isLoading: false,
	items: [],
	totalCount: 0,
};

export default (state = initialState, { type, payload }) => {
	switch (type) {
		case ActionTypes.FETCH_LESSON_REQUEST:
		case ActionTypes.FETCH_LESSON_LIST_REQUEST:
		case ActionTypes.CREATE_LESSON_REQUEST:
		case ActionTypes.UPDATE_LESSON_REQUEST:
		case ActionTypes.DELETE_LESSON_REQUEST:
			return {
				...state,
				isLoading: true,
			};
		case ActionTypes.FETCH_LESSON_FAILURE:
		case ActionTypes.FETCH_LESSON_LIST_FAILURE:
		case ActionTypes.CREATE_LESSON_FAILURE:
		case ActionTypes.UPDATE_LESSON_FAILURE:
		case ActionTypes.DELETE_LESSON_FAILURE:
			return {
				...state,
				isLoading: false,
			};
		case ActionTypes.CREATE_LESSON_SUCCESS:
		case ActionTypes.UPDATE_LESSON_SUCCESS:
		case ActionTypes.DELETE_LESSON_SUCCESS:
			return {
				...state,
				isLoading: false,
			};
		case ActionTypes.FETCH_LESSON_SUCCESS:
			return {
				...state,
				isLoading: false,
				items: [payload.lesson],
				totalCount: 1,
			};
		case ActionTypes.FETCH_LESSON_LIST_SUCCESS:
			return {
				...state,
				isLoading: false,
				items: [...payload.lessons],
				totalCount: payload.totalCount,
			};
		case ActionTypes.CLEAR_LESSON_LIST:
			return {
				...state,
				items: [],
				totalCount: 0,
			};
		default:
			return state;
	}
};
