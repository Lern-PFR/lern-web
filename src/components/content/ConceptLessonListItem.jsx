import { useCallback, useMemo } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { generatePath } from 'react-router';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';

import routes from 'routes';
import { history } from 'routes/components/RouterProvider';

import { StyledDiv, StyledListItem } from 'components/shared/styledElements';
import { BodyCopy, Brevier } from 'components/shared/typography';
import { OutlinedButton, OutlinedLinkButton } from 'components/shared/buttons';
import Link from 'components/shared/navigation/Link';

import { innerContentListItem, innerContentListItemActionsRow, moduleCard } from 'theme/contentEditionCommon/innerContentListTheme';
import { deleteLesson } from 'redux/actions/lessons';

const ListCard = styled(StyledListItem)(
	{
		...innerContentListItem,
		'&, & *': {
			cursor: 'pointer',
		},
	}
);

/**
 * @name ConceptLessonListItem
 * @description A component used to display a concept's lesson in the concept edition page.
 *
 * @author TimothÃ©e Simon-Franza
 *
 * @param {string} description	The current lesson's description.
 * @param {string} id			The current lesson's id.
 * @param {string} title		THe current lesson's title.
 */
const ConceptLessonListItem = ({ description, id, title }) => {
	const dispatch = useDispatch();
	const { t } = useTranslation();

	const editPageRedirectionLink = useMemo(() => generatePath(routes.lessons.lessonEdition, { lessonId: id }), [id]);

	const onCardClick = useCallback(() => {
		history.push(editPageRedirectionLink);
	}, [editPageRedirectionLink]);

	const onDeleteBtnClick = useCallback((e) => {
		if (e?.stopPropagation) {
			e.stopPropagation();
		}

		dispatch(deleteLesson(id));
	}, [dispatch, id]);

	return (
		<ListCard key={id} onClick={onCardClick} {...moduleCard}>
			<BodyCopy margin="0">{title}</BodyCopy>
			<Brevier margin="0">{description}</Brevier>
			<StyledDiv {...innerContentListItemActionsRow}>
				<OutlinedLinkButton>
					<Link to={editPageRedirectionLink}>{t('concepts.edition.lessons_list.links.edit')}</Link>
				</OutlinedLinkButton>
				<OutlinedButton onClick={onDeleteBtnClick}>
					{t('concepts.edition.lessons_list.links.delete')}
				</OutlinedButton>
			</StyledDiv>
		</ListCard>
	);
};

ConceptLessonListItem.propTypes = {
	description: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
};

export default ConceptLessonListItem;
