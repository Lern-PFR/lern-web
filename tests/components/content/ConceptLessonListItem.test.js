import { shallow } from 'enzyme';
import { act, fireEvent, render, screen } from '@testing-library/react';

import { Router, StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ConceptLessonListItem from 'components/content/ConceptLessonListItem';

import { ActionTypes } from 'redux/actions/lessons';
import { createMemoryHistory } from 'history';
import routes from 'routes';

describe('Concept edition page - lesson list item', () => {
	const mockStore = configureMockStore([thunk]);
	const store = mockStore({});
	const mockedLesson = { id: 'dummy_lesson_id_1', title: 'dummy_lesson_title_1', description: 'dummy_lesson_desc_1' };

	afterEach(() => {
		jest.restoreAllMocks();
		jest.resetAllMocks();
	});

	describe('snapshots', () => {
		it('should match previous snapshot', () => {
			const sut = (
				<Provider store={store}>
					<ConceptLessonListItem {...mockedLesson} conceptId="1" />
				</Provider>
			);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('onClick events', () => {
		it('should dispatch a lesson deletion request action upon click on the lesson card\'s delete button.', () => {
			render(
				<StaticRouter>
					<Provider store={store}>
						<ConceptLessonListItem {...mockedLesson} conceptId="1" />
					</Provider>
				</StaticRouter>
			);

			act(() => {
				fireEvent.click(screen.getByText('concepts.edition.lessons_list.links.delete'));
			});

			expect(store.getActions()).toEqual([{ type: ActionTypes.DELETE_LESSON_REQUEST }]);
		});

		it('should redirect to the lesson\'s edition page url upon click on the card\'s edit button.', () => {
			const history = createMemoryHistory();

			render(
				<Router history={history}>
					<Provider store={store}>
						<ConceptLessonListItem {...mockedLesson} conceptId="1" />
					</Provider>
				</Router>
			);

			act(() => {
				fireEvent.click(screen.getByText('concepts.edition.lessons_list.links.edit'));
			});

			expect(history.location.pathname).toEqual(routes.lessons.lessonEdition.replace(':lessonId', mockedLesson.id));
		});
	});
});
