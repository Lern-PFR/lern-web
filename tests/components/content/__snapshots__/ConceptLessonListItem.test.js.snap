// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concept edition page - lesson list item snapshots should match previous snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ConceptLessonListItem
    conceptId="1"
    description="dummy_lesson_desc_1"
    id="dummy_lesson_id_1"
    title="dummy_lesson_title_1"
  />
</ContextProvider>
`;
